### PR TITLE
Update infer-address-spaces.cpp

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

File: clang/test/CodeGenSYCL/infer-address-spaces.cpp [hunk 1]
```diff
@@ -17,5 +17,5 @@ void foo(const float *usm_in, float* usm_out) {
 
 // No addrspacecast before loading and storing values
 // CHECK-NOT: addrspacecast
-// CHECK:  %0 = load float, ptr addrspace(1)
-// CHECK:  store float %0, ptr addrspace(1)
+// CHECK:  %__SYCLKernel.ascast.val1.val = load float, ptr addrspace(1)
```
Comment (Fznamznon) [inline clang/test/CodeGenSYCL/infer-address-spaces.cpp:20]: To avoid the same problem in the future, how about using a regexp here?

Comment (FreddyLeaf) [reply clang/test/CodeGenSYCL/infer-address-spaces.cpp]: Agree, done in ed6d2e0e542751da0aba05421d9719ffe724c9ec

**Review by smanna12 — APPROVED** [review]

**Comment by FreddyLeaf** [comment]
Thank you both for review!

File: clang/test/CodeGenSYCL/infer-address-spaces.cpp [hunk 2]
```diff
@@ -17,5 +17,5 @@ void foo(const float *usm_in, float* usm_out) {
 
 // No addrspacecast before loading and storing values
 // CHECK-NOT: addrspacecast
-// CHECK:  %0 = load float, ptr addrspace(1)
-// CHECK:  store float %0, ptr addrspace(1)
+// CHECK:  {{%[^ ]*}} = load float, ptr addrspace(1)
+// CHECK:  store float {{%[^ ]*}}, ptr addrspace(1)
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/CodeGenSYCL/infer-address-spaces.cpp:21]: >> I think to ensure that, we probably need to use a variable and make sure the same thing is used in store.

Agreed. I missed that. 

@FreddyLeaf could you please address @Fznamznon's comment? Thank you

[the remark this answers is not in the document]
Comment (againull) [reply clang/test/CodeGenSYCL/infer-address-spaces.cpp:21]: Applied follow up fix here: https://github.com/intel/llvm/commit/d918ff92b45e8f471b3b9cd17255f3d99b2ac757

[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/CodeGenSYCL/infer-address-spaces.cpp:21]: Thanks @againull

[the remark this answers is not in the document]
Comment (FreddyLeaf) [reply clang/test/CodeGenSYCL/infer-address-spaces.cpp:21]: Thanks @againull for the follow up fix!

**Pull request merged**